src/main/java/com/sprint/mission/discodeit/dto/data/DiscodeitUserDetails.java

```diff
@@ -0,0 +1,68 @@
+package com.sprint.mission.discodeit.dto.data;
+
+import com.sprint.mission.discodeit.entity.BinaryContent;
+import com.sprint.mission.discodeit.entity.User;
+import com.sprint.mission.discodeit.mapper.UserMapper;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+@Getter
+@RequiredArgsConstructor
+public class DiscodeitUserDetails implements UserDetails {
+
+  private final UserDto userDto;
+  private final String password;
+
+  @Override
+  public Collection<? extends GrantedAuthority> getAuthorities() {
+    if (userDto == null) {
+      return List.of();
+    }
+
+    return List.of(new SimpleGrantedAuthority(userDto.role().name()));
```

Comment:
Spring Security에서는 `ROLE_`  prefix를 앞에 꼭 붙여아 동작해요! 그래야 SecurityConfig의 hasRole("ADMIN") 체크가 올바르게 작동합니다.

```suggestion
    return List.of(new SimpleGrantedAuthority("ROLE_" + userDto.role().name()));
```

- [공식문서](https://docs.spring.io/spring-security/reference/servlet/authorization/architecture.html)
> By default, role-based authorization rules include **ROLE_** as a prefix. This means that if there is an authorization rule that requires a security context to have a role of "USER", Spring Security will by default look for a GrantedAuthority#getAuthority that returns "ROLE_USER".




```java
public final class AuthorizeHttpRequestsConfigurer<H extends HttpSecurityBuilder<H>>
    
	private String rolePrefix = "ROLE_";

}
```